src/qtree.js

```diff
@@ -0,0 +1,436 @@
+/**
```

Comment:
It was still giving me trouble so I just reverted back to my crappy fix until later so I could get back to physics stuff.